prov/efa/test/efa_unit_test_cq.c

```diff
@@ -286,4 +287,122 @@ void test_rdm_cq_read_failed_poll()
 }
 
+#if HAVE_EFADV_CQ_EX
+static void rdm_cq_read_unknown_peer_ah(int peer_removed) {
+	struct efa_cq *efa_cq;
+	struct rxr_ep *rxr_ep;
+	struct efa_resource resource = {0};
+	struct rxr_pkt_entry *entry;
+	struct efa_ep_addr raw_addr = {0};
+	size_t raw_addr_len = sizeof(raw_addr);
+	struct rdm_peer *peer;
+	struct fi_cq_data_entry cq_entry;
+	struct fi_cq_err_entry cq_err_entry;
+	struct rxr_eager_msgrtm_hdr base_hdr = {0};
+	struct rxr_req_opt_connid_hdr opt_connid_hdr = {0};
+	uint32_t *connid;
+	struct efadv_cq efadv_cq = {0};
+	int ret;
+
+	efa_unit_test_resource_construct(&resource, FI_EP_RDM);
+
+	rxr_ep = container_of(resource.ep, struct rxr_ep, util_ep.ep_fid);
+	efa_cq = container_of(rxr_ep->rdm_cq, struct efa_cq, util_cq.cq_fid);
+
+	/* Create and register a fake peer */
+	ret = fi_getname(&resource.ep->fid, &raw_addr, &raw_addr_len);
+	assert_int_equal(ret, 0);
+	raw_addr.qpn = 0; // return value of _read_qp_num
+	raw_addr.qkey = 0x1234;
+
+	if (!peer_removed) {
+		fi_addr_t peer_addr;
+		ret = fi_av_insert(resource.av, &raw_addr, 1, &peer_addr, 0, NULL);
+		assert_int_equal(ret, 1);
+	}
+
+	/* Setup packet entry */
+	entry = rxr_pkt_entry_alloc(rxr_ep, rxr_ep->efa_rx_pkt_pool, RXR_PKT_FROM_EFA_RX_POOL);
+	base_hdr.hdr.type = RXR_EAGER_MSGRTM_PKT; // packet type must be in [RXR_REQ_PKT_BEGIN, RXR_EXTRA_REQ_PKT_END)
+	base_hdr.hdr.flags |= RXR_PKT_CONNID_HDR | RXR_REQ_MSG;
+	base_hdr.hdr.msg_id = 0;
+	memcpy(entry->pkt, &base_hdr, sizeof(struct rxr_eager_msgrtm_hdr));
+	assert_int_equal(rxr_get_base_hdr(entry->pkt)->type, RXR_EAGER_MSGRTM_PKT);
+	assert_int_equal(rxr_pkt_req_base_hdr_size(entry), sizeof(struct rxr_eager_msgrtm_hdr));
+	opt_connid_hdr.connid = raw_addr.qkey;
+	memcpy(entry->pkt + sizeof(struct rxr_eager_msgrtm_hdr), &opt_connid_hdr, sizeof(struct rxr_req_opt_connid_hdr));
+	connid = rxr_pkt_connid_ptr(entry);
+	assert_int_equal(*connid, raw_addr.qkey);
+
+	if (!peer_removed) {
+		/* Skip handshake */
+		peer = rxr_ep_get_peer(rxr_ep, entry->addr);
+		assert_non_null(peer);
+		peer->flags |= RXR_PEER_HANDSHAKE_SENT;
+	}
+
+	/* Setup CQ */
+	efa_cq->ibv_cq_ex->wr_id = (uintptr_t)entry;
+	efa_cq->ibv_cq_ex->start_poll = &efa_mock_ibv_start_poll_return_mock;
+	efa_cq->ibv_cq_ex->next_poll = &efa_mock_ibv_next_poll_return_mock;
+	efa_cq->ibv_cq_ex->end_poll = &efa_mock_ibv_end_poll_return_mock;
+	efa_cq->ibv_cq_ex->read_slid = &efa_mock_ibv_read_slid_return_mock;
+	efa_cq->ibv_cq_ex->read_byte_len = &efa_mock_ibv_read_byte_len_return_mock;
+	efa_cq->ibv_cq_ex->read_opcode = &efa_mock_ibv_read_opcode_return_mock;
+	efa_cq->ibv_cq_ex->read_src_qp = &efa_mock_ibv_read_src_qp_return_mock;
 
+	efadv_cq.wc_read_ah = &efa_mock_efadv_wc_read_unknown_ah;
+
+	/* Read 1 entry with unknown AH */
+	will_return(efa_mock_ibv_start_poll_return_mock, 0);
+	will_return(efa_mock_ibv_next_poll_return_mock, ENOENT);
+	will_return(efa_mock_ibv_end_poll_return_mock, NULL);
+	will_return(efa_mock_ibv_read_slid_return_mock, 0xffff);
```

Comment:
Add a comment here that `slid` being `0xffff` means device does not know the address.